src/module/tests/DefenseTest.ts

```diff
@@ -60,4 +69,72 @@ export class DefenseTest<T extends DefenseTestData = DefenseTestData> extends Op
     get initiativeModifier(): number {
         return this.data.iniMod || 0;
     }
-}
\ No newline at end of file
+
+    override canConsumeDocumentResources() {
+        // Check if the actor is in active combat situation and has enough initiative score left.
+        if (this.actor && this.data.iniMod && game.combat) {
+            const combat: SR5Combat = game.combat as unknown as SR5Combat;
+            const combatant = combat.getActorCombatant(this.actor);
+            if (!combatant?.initiative) return true;
+
+            if (combatant && combatant.initiative + this.data.iniMod < 0) {
+                ui.notifications?.warn('SR5.MissingRessource.Initiative', {localize: true});
+                return false;
+            }
+        }
+
+        return super.canConsumeDocumentResources();
+    }
+
+    /**
+     * Based in combatants ini score, pre-filter available active defense modes.
+     *
+     * This behaviour can be disabled using the must have ressources setting.
+     */
+    _filterActiveDefenses() {
+        if (!this.actor) return;
+
+        // Don't validate ini costs when costs are to be ignored.
+        const mustHaveRessouces = game.settings.get(SYSTEM_NAME, FLAGS.MustHaveRessourcesOnTest);
+        if (!mustHaveRessouces) return;
+
+        // TODO: Check ressource setting.
+        const iniScore = this.actor.combatInitiativeScore;
+        Object.values(this.data.activeDefenses).forEach(mode =>
+            { mode.disabled = CombatRules.canUseActiveDefense(iniScore, mode.initMod) }
+        )
+    }
+
+    /**
+     * Should an active defense be selected apply the initiative modifier to the defenders combat initiative.
+     */
+    applyIniModFromActiveDefense() {
```

Comment:
Can this be viewed as a new feature outside of matrix as well? It's used _for_ MatrixDefense but I'm not sure if this is fully new or just refactored.

If so, we can add it to the https://github.com/SR5-FoundryVTT/SR5-FoundryVTT/pull/1292 PR feature list.